openmc/model/model.py

```diff
@@ -151,44 +219,152 @@ def from_xml(cls, geometry='geometry.xml', materials='materials.xml',
 
-    def deplete(self, timesteps, chain_file=None, method='cecm',
-                fission_q=None, **kwargs):
+    def init_lib(self, threads=None, geometry_debug=False, restart_file=None,
+                 tracks=False, output=True, event_based=None):
+        """Initializes the model in memory via the C API
+
+        .. versionadded:: 0.13.0
+
+        Parameters
+        ----------
+        threads : int, optional
+            Number of OpenMP threads. If OpenMC is compiled with OpenMP
+            threading enabled, the default is implementation-dependent but is
+            usually equal to the number of hardware threads available
+            (or a value set by the :envvar:`OMP_NUM_THREADS` environment
+            variable).
+        geometry_debug : bool, optional
+            Turn on geometry debugging during simulation. Defaults to False.
+        restart_file : str, optional
+            Path to restart file to use
+        tracks : bool, optional
+            Write tracks for all particles. Defaults to False.
+        output : bool
+            Capture OpenMC output from standard out
+        event_based : None or bool, optional
+            Turns on event-based parallelism if True. If None, the value in
+            the Settings will be used.
+        """
+
+        # TODO: right now the only way to set most of the above parameters via
+        # the C API are at initialization time despite use-cases existing to
+        # set them for individual runs. For now this functionality is exposed
+        # where it exists (here in init), but in the future the functionality
+        # should be exposed so that it can be accessed via model.run(...)
+
+        args = openmc.process_CLI_arguments(
+            volume=False, geometry_debug=geometry_debug,
+            restart_file=restart_file, threads=threads, tracks=tracks,
+            event_based=event_based)
+        # Args adds the openmc_exec command in the first entry; remove it
+        args = args[1:]
+
+        self.finalize_lib()
+
+        if self.intracomm.rank == 0:
+            self.export_to_xml()
+        self.intracomm.barrier()
+
+        if isinstance(self.intracomm, openmc.DummyCommunicator):
+            # openmc.lib.init does not accept DummyCommunicator, and importing
+            # the DummyCommunicator class there is overkill. Filter it here
+            intracomm = None
+        else:
+            intracomm = self.intracomm
+        openmc.lib.init(args=args, intracomm=intracomm, output=output)
+
+    def finalize_lib(self):
+        """Finalize simulation and free memory allocated for the C API
+
+        .. versionadded:: 0.13.0
+
+        """
+
+        openmc.lib.finalize()
+
+    def deplete(self, timesteps, method='cecm', final_step=True,
+                operator_kwargs=None, directory='.', output=True,
+                **integrator_kwargs):
         """Deplete model using specified timesteps/power
 
+        .. versionchanged:: 0.13.0
```

Comment:
```suggestion
        .. versionchanged:: 0.13.0
           The *final_step*, *operator_kwargs*, *directory*, and *output*
           arguments were added.
```

Reply:
And I just learned that ``versionchanged`` requires a description whereas ``versionadded`` does not! (first time using is right in this PR!)